Common/Data/Market/Tick.cs

```diff
@@ -47,10 +49,37 @@ public class Tick : BaseData
         public decimal Quantity = 0;
 
         /// <summary>
-        /// Exchange we are executing on. String short code expanded in the MarketCodes.US global dictionary
+        /// Exchange code this tick came from <see cref="Exchanges"/>
         /// </summary>
         [ProtoMember(12)]
-        public string Exchange = "";
+        public byte ExchangeCode
+        {
+            get
+            {
+                return _exchangeCode;
+            }
+            set
+            {
+                _exchangeCode = value;
+                _exchange = ((PrimaryExchange) value).ToString();
```

Comment:
> _exchange = ((PrimaryExchange) value).ToString();

Cool! let's add a test where value isn't in `PrimaryExchange` enum, think we should handle this case 👍